hubstorage/client.py

```diff
@@ -21,22 +42,48 @@ class HubstorageClient(object):
     DEFAULT_TIMEOUT = 60.0
 
     def __init__(self, auth=None, endpoint=None, connection_timeout=None,
-            max_retries=0):
+            max_retries=3):
         self.auth = xauth(auth)
         self.endpoint = endpoint or self.DEFAULT_ENDPOINT
         self.connection_timeout = connection_timeout or self.DEFAULT_TIMEOUT
-        self.session = self._create_session(max_retries)
+        self.session = self._create_session()
+        self.retrier = Retrying(stop_max_attempt_number=max_retries + 1, retry_on_exception=_hc_retry_on_exception)
```

Comment:
I understand you are trying to keep the patch simple, but if we go for Retrying what do you think if it exposes retrying options like delays, randomness and exponential backoff. 

Let's choose conservative but still useful defaults, like enabling exponential backoff with `stop_max_delay` set to DEFAULT_TIMEOUT and `wait_exponential_multiplier` tuned for 3 retries total.

I am not sure what is the most practical way to expose retrying options, but as a minimum use a method to construct it so users can override to suit their needs.